src/java.desktop/share/classes/com/sun/imageio/plugins/common/ImageUtil.java

```diff
@@ -1162,30 +1160,17 @@ public static final boolean imageIsContiguous(RenderedImage image) {
     }
 
     /**
-     * Returns <code>true</code> if the given <code>ColorSpace</code> object is
-     * an instance of <code>ICC_ColorSpace</code> but is not one of the standard
-     * <code>ColorSpace</code>s returned by <code>ColorSpace.getInstance()</code>.
+     * Returns {@code true} if the given {@code ColorSpace} object is an
+     * instance of {@code ICC_ColorSpace} but is not one of the standard
+     * {@code ColorSpace}s returned by {@code ColorSpace.getInstance()}.
      *
-     * @param cs The <code>ColorSpace</code> to test.
+     * @param  cs the {@code ColorSpace} to test
      */
     public static boolean isNonStandardICCColorSpace(ColorSpace cs) {
-        boolean retval = false;
-
-        try {
-            // Check the standard ColorSpaces in decreasing order of
-            // likelihood except check CS_PYCC last as in some JREs
-            // PYCC.pf used not to be installed.
-            retval =
-                (cs instanceof ICC_ColorSpace) &&
-                !(cs.isCS_sRGB() ||
-                  cs.equals(ColorSpace.getInstance(ColorSpace.CS_LINEAR_RGB)) ||
-                  cs.equals(ColorSpace.getInstance(ColorSpace.CS_GRAY)) ||
-                  cs.equals(ColorSpace.getInstance(ColorSpace.CS_CIEXYZ)) ||
-                  cs.equals(ColorSpace.getInstance(ColorSpace.CS_PYCC)));
-        } catch(IllegalArgumentException e) {
-            // PYCC.pf not installed: ignore it - 'retval' is still 'false'.
-        }
-
-        return retval;
+        return cs instanceof ICC_ColorSpace && !cs.isCS_sRGB()
+                && !cs.equals(ColorSpace.getInstance(ColorSpace.CS_LINEAR_RGB))
+                && !cs.equals(ColorSpace.getInstance(ColorSpace.CS_GRAY))
+                && !cs.equals(ColorSpace.getInstance(ColorSpace.CS_CIEXYZ))
+                && !cs.equals(ColorSpace.getInstance(ColorSpace.CS_PYCC));
     }
```

Comment:
The method above was added by the tiff implementation in jdk9 and it is mostly a copy of the older code in the jpeg plugin. I merged two methods and leave it here in the "common" code.